experimental/list_formatter/src/list_formatter.rs

```diff
@@ -84,8 +86,12 @@ impl ListFormatter {
     pub fn format(&self, values: &[&str]) -> String {
```

Comment:
Nit: Now that you have size_hint, you should implement `Writeable`, change `format` to `format_to_string`, and rewire it to call the `Writeable` impl.

We should do this before upgrading ListFormatter to components.

Reply:
Hmm, this is tricky. The ICU spec says that the list output has to be generated from back to front, which is not possible with Writeable. Generating front-to-back will only be equivalent under some subset of patterns, I need to verify that this is true for the complete CLDR data.

Reply:
Acknowledged; let's think about this but not block the current PR